node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContextImpl.java

```diff
@@ -209,8 +229,8 @@ public Builder zone(ZoneApi zone) {
             return this;
         }
 
-        public Builder pathToContainerStorage(Path pathToContainerStorage) {
-            this.pathToContainerStorage = pathToContainerStorage;
+        public Builder pathToContainerStorageFromFileSystem(FileSystem fileSystem) {
```

Comment:
So is this still needed? Ideally, if the filesystem is set in the builder, both paths should be mapped to that fs in `build()` and this method should left as-is.

Reply:
This diff is somewhat deceiving: I have renamed `fileSystem(FileSystem)` to `pathToContainerStorageFromFileSystem(FileSystem)` as the old method name was misleading: It only set the container storage path.

Then I added a new `fileSystem(FileSystem)` which sets the file system to be used when resolving default paths, unless these have been overridden with the `pathToContainerStorageFromFileSystem` and `pathToVespaHome(Path)` methods.

Reply:
The old method let you specify whatever pah you wanted for container storage, and filesystem was a separate thing. With the new method there is no way to not use /home/docker as container storage

Reply:
The original method was removed because it was unused.

Reply:
So why was this method needed? Wouldn't the change to `fileSystem()` and `build()` have the same effect?

Reply:
This method is the old `fileSystem(FileSystem)` method which is used in various tests. The only thing it did was to set the container storage path, so I have given it a better name.